### PR TITLE
[org-only] relax org name restriction

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Relaxes the unnecessary strict requirments for organization names.

## Related Issue(s)

Fixes WEB-220

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-org-nam7ae9c219c3</li>
	<li><b>🔗 URL</b> - <a href="https://se-org-nam7ae9c219c3.preview.gitpod-dev.com/workspaces" target="_blank">se-org-nam7ae9c219c3.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/gitpod-db/src/user-to-team-migration-service.ts [hunk 1]
```diff
@@ -72,9 +75,6 @@ export class UserToTeamMigrationService {
         if (!team) {
             throw new ResponseError(ErrorCodes.CONFLICT, "Could not create team for user.", { userId: user.id });
         }
-        // add membership
-        await this.teamDB.addMemberToTeam(user.id, team.id);
-        await this.teamDB.setTeamMemberRole(user.id, team.id, "owner");
```
Comment (easyCZ) [inline components/gitpod-db/src/user-to-team-migration-service.ts:77]: What's the reason for this removal?

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:77]: Sorry, wanted to add a comment. This is all done as part of createTeam

File: components/gitpod-db/src/typeorm/team-db-impl.ts [hunk 2]
```diff
@@ -188,15 +188,29 @@ export class TeamDBImpl implements TeamDB {
...
+        name = name.trim();
+        if (name.length < 3) {
             throw new ResponseError(
                 ErrorCodes.BAD_REQUEST,
-                "Please choose a name containing only letters, numbers, -, _, ', or spaces.",
+                "Please choose a name that is at least three characters long.",
             );
         }
+        // check that the trimmed name is at most 32 characters long
+        if (name.length > 32) {
```
[the remark this answers is not in the document]
Comment (easyCZ) [reply components/gitpod-db/src/typeorm/team-db-impl.ts:200]: 64/128 chars but I don't mind, just didn't have context

**Review by easyCZ — APPROVED** [review]

File: components/gitpod-db/src/typeorm/team-db-impl.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/gitpod-db/src/typeorm/team-db-impl.ts]: 👍  I changed it to 64

[the remark this answers is not in the document]
Comment (easyCZ) [reply components/gitpod-db/src/typeorm/team-db-impl.ts]: Thinking through this again, I think 63 or 127 would've been better choices, given it doesn't require an extra memory word to store

**Pull request merged**